src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/TelemetryNotification.cs

```diff
@@ -0,0 +1,54 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using Microsoft.SqlTools.ServiceLayer.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageServices.Contracts
+{
+    /// <summary>
+    /// Parameters sent back with an IntelliSense ready event
+    /// </summary>
+    public class TelemetryParams
+    {
+        public string EventName { get; set; }
+
+        /// <summary>
+        /// URI identifying the text document
+        /// </summary>
+        public Dictionary<string, string> Properties { get; set; }
+
+        /// <summary>
+        /// URI identifying the text document
+        /// </summary>
+        public Dictionary<string, double> Measures { get; set; }
+    }
+
+    /// <summary>
+    /// Event sent when the language service needs to add a telemetry event
+    /// </summary>
+    public class TelemetryNotification
+    {
+        public static readonly
+            EventType<TelemetryParams> Type =
+            EventType<TelemetryParams>.Create("textDocument/telemetry");
```

Comment:
>textDocument/telemetry [](start = 47, length = 22)

did we consider using the language service protocol "telemetry/event" event type?  [https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#telemetry_event](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#telemetry_event)